### PR TITLE
Upgrade to Gradle 7.3.3

### PR DESCRIPTION
**Pull request by jjaakola-aiven**
Gradle upgrade to 7.3.3 and build script update.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,7 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=b13f5d97f08000996bf12d9dd70af3f2c6b694c2c663ab1b545e9695562ad1ee
```
Comment (ivanyu) [inline gradle/wrapper/gradle-wrapper.properties:3]: `distributionSha256Sum` must be updated. We use it to ensure the integrity of Gradle distributions.

**Review by ivanyu — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jjaakola-aiven) [reply gradle/wrapper/gradle-wrapper.properties:3]: Wrapper upgrade removed this.

Comment (ivanyu) [reply gradle/wrapper/gradle-wrapper.properties:3]: I believe it's this value:
https://services.gradle.org/distributions/gradle-7.3.3-all.zip.sha256

**Review by juha-aiven — COMMENTED** [review]
I tried 

```
./gradlew wrapper --gradle-version 7.3.3 --gradle-distribution-sha256-sum=b586e04868a22fd817c8971330fec37e298f3242eb85c374181b12d637f80302
```

And that generates the same wrapper files.

File: build.gradle [hunk 2]
```diff
@@ -59,7 +59,8 @@ sourceSets {
     integrationTest {
         java.srcDir file('src/integration-test/java')
         resources.srcDir file('src/integration-test/resources')
-        compileClasspath += sourceSets.main.output + configurations.testRuntime
+        configurations.testRuntimeOnly.setCanBeResolved(true)
```
[the remark this answers is not in the document]
Comment (jjaakola-aiven) [reply build.gradle:62]: The configuration is not resolvable, I do not remember how the Gradle APIs did change between 6.x and 7.x exactly. In any case the configuration `runtime` has been changed to `runtimeOnly´, applies also to `<sourceSet>Runtime` -> `<sourceSet>RuntimeOnly`.

I'll test if this works with `runtimeClasspath` configuration, it is resolvable.

[the remark this answers is not in the document]
Comment (jjaakola-aiven) [reply build.gradle]: The `configuration.testRuntimeClasspath` is the correct configuration.

**Review by juha-aiven — APPROVED** [review]
LGTM

**Pull request merged**